components/tools/OmeroWeb/omeroweb/webgateway/static/webgateway/js/ome.roidisplay.js

```diff
@@ -242,18 +252,18 @@ $.fn.roi_display = function(options) {
             }
             return {'x':sel_x, 'y':sel_y};
         }
-        
+
         // called when user clicks on ROI
         handle_shape_click = function(event) {
             var shape = this;
-            var shape_id = parseInt(shape.id);
+            var shape_id = resolve_id(shape.id);
             self.set_selected_shape(shape_id);
         }
 
         // load the ROIs from json call and display
-        load_rois = function(display_rois, filter) {
+        load_rois = function(display_rois, filter, roi_id, shape_id, shape_config, refr_rois, hide_ome_rois) {
```

Comment:
Perhaps it would be nicer to simply supply a "callback" argument here instead of adding all the parameters needed to add an ROI. This would make it amenable to other actions we'd want.
E.g. 
```
load_rois = function(display_rois, filter, callback) {

...
   if (callback) {
    callback();
}
```
Then use it like this:

```
load_rois(rois, undefined, function() {
   $viewportimg.trigger("push_shape", [roi_id, shape_id, shape_config, refr_rois, hide_ome_rois]);
});
```